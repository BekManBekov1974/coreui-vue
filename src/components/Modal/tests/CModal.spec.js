import { mount } from '@vue/test-utils'
import Component from '../CModal'

const ComponentName = 'CModal'
const defaultWrapper = mount(Component)
const customWrapper = mount(Component, {
  propsData: {
    visible: true,
    centered: true,
    title: 'modal title',
    size: 'lg',
    variant: 'success',
    borderVariant: 'success',
    noFade: true,
    noBackdrop: true,
    noCloseOnBackdrop: true,
    noFooter: true,
    addModalClasses: 'additional-modal-class',
    addDialogClasses: 'additional-dialog-class',
    addContentClasses: 'additional-content-class'
  },
  slots: {
    default: 'CModal body'
  }
})

describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
  it('renders correctly', () => {
    expect(customWrapper.element).toMatchSnapshot()
  })
})
