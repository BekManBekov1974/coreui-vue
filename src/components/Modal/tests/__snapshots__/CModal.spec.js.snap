// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CModal renders correctly 1`] = `
<div>
  <div
    class="c-modal c-overflow-auto c-fade"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="c-modal-dialog"
      role="document"
    >
      <div
        class="c-modal-content"
      >
        <div
          class="c-modal-header"
        >
          <h5
            class="c-modal-title"
          >
            
                
              
          </h5>
           
          <button
            aria-label="Close"
            class="c-close"
            type="button"
          >
            <span>
              ×
            </span>
          </button>
        </div>
         
        <div
          class="c-modal-body"
        />
         
        <div
          class="c-modal-footer"
        >
          <button
            class="c-btn c-btn-secondary"
            type="button"
          >
            
              Cancel
            
          </button>
           
          <button
            class="c-btn c-btn-primary"
            type="button"
          >
            
              OK
            
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <!---->
</div>
`;

exports[`CModal renders correctly 2`] = `
<div>
  <div
    class="c-modal c-overflow-auto additional-modal-class c-show c-d-block c-modal-success"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="additional-dialog-class c-modal-dialog c-modal-dialog-centered c-modal-lg"
      role="document"
    >
      <div
        class="additional-content-class c-modal-content c-border-success"
      >
        <div
          class="c-modal-header"
        >
          <h5
            class="c-modal-title"
          >
            
                modal title
              
          </h5>
           
          <button
            aria-label="Close"
            class="c-close"
            type="button"
          >
            <span>
              ×
            </span>
          </button>
        </div>
         
        <div
          class="c-modal-body"
        >
          CModal body
        </div>
         
        <!---->
      </div>
    </div>
  </div>
   
  <!---->
</div>
`;
